dataplane/standalone/saiserver/routing.go

```diff
@@ -238,6 +242,31 @@ func (r *route) CreateRouteEntry(ctx context.Context, req *saipb.CreateRouteEntr
 		case saipb.ObjectType_OBJECT_TYPE_ROUTER_INTERFACE:
 			rReq.Route.Hop = &dpb.Route_InterfaceId{InterfaceId: fmt.Sprint(req.GetNextHopId())}
 		case saipb.ObjectType_OBJECT_TYPE_PORT:
+			attrReq := &saipb.GetSwitchAttributeRequest{
+				Oid:      req.GetEntry().GetSwitchId(),
+				AttrType: []saipb.SwitchAttr{saipb.SwitchAttr_SWITCH_ATTR_CPU_PORT},
+			}
+			resp := &saipb.GetSwitchAttributeResponse{}
+			if err := r.mgr.PopulateAttributes(attrReq, resp); err != nil {
+				return nil, err
+			}
```

Comment:
Since this is unconditionally a CPU port then move this inside the if?

Reply:
good catch, this should be an early return, not unconditional